Week 3: 권시훈 Pull Request

kanai1 opened the pull request:
Feat: 스터디 실습 코드
Feat: 과제

kanai1 committed:
Merge pull request #2 from KGU-C-Lab/master

Week 2: Pull Request from C-lab

kanai1 committed:
Week 3: 권시훈 스터디파일 제출

Feat: 권시훈 스터디 파일 제출
- 스터디 코드 및 주석 제출

kanai1 committed:
Week 3: 권시훈 1번 과제 제출

Feat: 회원가입 UI 제작
- "SIGN UP" 간판
- 아이디, 비밀번호, 닉네임, MBTI, 한줄소개 입력칸
- 회원가입하기 버튼

kanai1 committed:
Week 3: 권시훈 2번 과제 제출

Feat: 계산기 앱 만들기
- 숫자 입력칸 2개
- 각 사칙연산에 해당하는 버튼 4개
- 출력 TextView
- 0으로 나눌경우 예외처리
- 숫자칸이 비어있는 경우 예외처리

chattymin commented on 권시훈/Week 3/과제/Calculator/app/src/main/java/com/kanai1/calculator/MainActivity.kt:
```diff
@@ -0,0 +1,89 @@
+        setContentView(binding.root)
+
+        binding.buttonAdd.setOnClickListener{
+            if(binding.input1.text.toString() == "" || binding.input2.text.toString() == "") {
```
내장함수를 활용하는 것도 좋은 방법일 것 같습니다.
그리고 내부에 공백이 들어있는 경우에도 에러가 발생할 것 같은데 이것도 고려하면 좋을 것 같네요
" "

chattymin commented on 권시훈/Week 3/과제/Calculator/app/src/main/java/com/kanai1/calculator/MainActivity.kt:
```diff
@@ -0,0 +1,89 @@
+        }
+
+        binding.buttonMinus.setOnClickListener{
+            if(binding.input1.text.toString() == "" || binding.input2.text.toString() == "") {
+                Toast.makeText(
+                    this,
+                    "숫자가 비어있습니다",
+                    Toast.LENGTH_SHORT
+                ).show()
+                binding.result.text = "Result: -"
```
이 내용은 대부분의 버튼 클릭시 중복되는데 함수화하면 좋을 것 같습니다!!

chattymin commented on 권시훈/Week 3/과제/signup/app/src/main/res/layout/activity_main.xml:
```diff
@@ -0,0 +1,133 @@
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent" />
+
+    <EditText
+        android:id="@+id/editTextID"
+        android:layout_width="0dp"
+        android:layout_height="wrap_content"
+        android:layout_marginTop="50dp"
+        android:ems="10"
+        android:inputType="text"
+        android:hint="아이디를 입력해주세요"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@+id/textView" />
+
+    <TextView
+        android:id="@+id/textView2"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:text="ID"
+        app:layout_constraintBottom_toTopOf="@+id/editTextID"
+        app:layout_constraintStart_toStartOf="@+id/editTextID" />
```
코드가 깔끔하고 좋네요!
그런데 ui상의 순서대로 있으면 가독성 측면에서 더 좋을 것 같습니다.ㅣ
그리고 변수 명이 조금 더 명확했으면 좋을 것 같네요

chattymin reviewed (changes requested):
이번주도 고생하셨습니다 :)

kanai1 committed:
Week 3: 권시훈 1번과제 수정사항 제출

DOSC: activity_main의 코드 수정사항 적용
- 컴포넌트의 UI순서와 코드상 순서가 같도록 조정
- 컴포넌트ID를 컨벤션에 맞게 조정

kanai1 committed:
Week 3: 권시훈 2번과제 수정사항 제출

Dosc: activity_main의 코드 수정사항 제출
- 코딩컨벤션에 맞춰 컴포넌트ID수정
- 이에따라 MainActivity에서도 변경
Feat: 에러처리코드 변경
- 중복되는 Toast와 Result설정부분 메소드로 처리
- editText가 비어있는지 확인하는 부분 isEmpty메소드 사용

kanai1 commented:
수정사항들 전부 확인 및 수정하고 Commit했습니다!

chattymin reviewed (approved):
깔끔한 코드 잘 봤습니다!!
아주그냥 야무져요~ :)
